Ruslan / Transaction export API

vantuz-subhuman opened the pull request:
1. Created separate `ExportApi` that can be used abstractly
2. Also created separate function to send any file as a download to user browser (as utils)
3. Created some API-boundary types to exchange data from any other part of the system into export-api

Export for all ADA transactions may be called as:
```
const rows = await yoroi.api.ada.getTransactionRowsToExport();
yoroi.api.export.exportTransactions({ rows, fileName: '<some_name>' });
```

![image](https://user-images.githubusercontent.com/5585355/50605324-11bd0780-0ed3-11e9-9691-77491ad377ef.png)

Tests for formatting function will be added.

nicarq commented on app/api/ada/index.js:
```diff
@@ -557,9 +559,45 @@ export default class AdaApi {
+    return transactions
+      .filter(tx => tx.ctCondition === 'CPtxInBlocks')
+      .map(tx => {
+        const fullValue = parseInt(tx.ctAmount.getCCoin, 10);
```
I think this overflows, so probably we should use `BigInt`

vantuz-subhuman replied:
Oh that's a really good catch, yeah. Replaced with `BigNumber` 👌

nicarq commented on app/api/export/index.js:
```diff
@@ -0,0 +1,135 @@
+}
+
+/**
+ * This api layer provides access to the electron local storage
```
electron -> browser? :D

vantuz-subhuman replied:
Damned copy-paste 😬 Fixed the whole comment )

nicarq commented on app/api/export/index.js:
```diff
@@ -0,0 +1,135 @@
+  }
+}
+
+function _writeCsvDataIntoCsvFile(data: CsvData): string {
```
I don’t know if it’s too picky, but instead of using `write` maybe we could use something different -- because it doesn’t directly write somewhere, but rather format --

vantuz-subhuman replied:
Yeah, I agree. Renamed and changed it a bit.

nicarq commented on app/api/export/utils.js:
```diff
@@ -0,0 +1,16 @@
+/**
+ * Make browser to download the specified blob of bytes as a file with the specified name
+ */
+export function sendFileToUser(data: Blob, fileName: string) {
```
good idea to put this in utils!

nicarq replied:
maybe we should make it async void?

vantuz-subhuman replied:
Yep, changed to async 👌

vantuz-subhuman commented on app/api/export/index.js:
```diff
@@ -0,0 +1,135 @@
+   * - rows: array of export-data
+   * - fileName: only name without extension
+   *
+   * No result will be returned. File is sent to user as side-effect.
```
Yep, changed to async 👌

nicarq commented:
LGTM :D

vantuz-subhuman commented:
@nicarq, @sarveshwar-singh, I have added some final changes:
1. Extracted some function into utils: https://github.com/Emurgo/yoroi-frontend/pull/210/files#diff-e05e93fdfc07fce3b2d6a5b92083def5
2. Created test-config to be used in tests: https://github.com/Emurgo/yoroi-frontend/pull/210/files#diff-a2e926e9dcc51b770ec51e30643a9a5b
3. Added unit-tests for these new functions in utils: https://github.com/Emurgo/yoroi-frontend/pull/210/files#diff-b45881d223bc76e38c5909a06d0cf197

Ready for review and merge

nicarq commented:
awesome! we have jest now :D
